### ISSUE TITLE
Multiple issues with single images

### ISSUE THREAD

**Issue by pyahmed** [mutschler/mt#45]
Command:
`mt.exe input.mp4`
Result:
`input.jpg`

Command:
`mt.exe --skip-existing input.mp4`
Result:
`input.jpg` is not generated again.

Command:
`mt.exe -s input.mp4`
Result:
```
input-01.jpg
input-02.jpg
input-03.jpg
input-04.jpg
```

Command:
`mt.exe --skip-existing -s input.mp4`
Result:
```
input-01.jpg (overwritten)
input-02.jpg (overwritten)
input-03.jpg (overwritten)
input-04.jpg (overwritten)
```

Command:
`mt.exe -s input.mp4`
Result:
```
input-01.jpg
input-02.jpg
input-03.jpg
input-04.jpg
input-05.jpg (new, but same timestamp as 01)
input-06.jpg (new, but same timestamp as 02)
input-07.jpg (new, but same timestamp as 03)
input-08.jpg (new, but same timestamp as 04)
```

Three issues:
1) So `--skip-existing` is ignored or does not apply to single images, which would be extremely handy if it did :-). Is there a way to have it also obey the `--skip-existing` flag?

2) If `--skip-existing` is not specified mt will just generate new files at the same timestamps but with consecutive numbers. If you run it again and again, it will make new files. When in addition you specify `-s` (single images) it will always generate (and overwrite) existing single images.

3) No vtt file is created when using in combination with single images (neither with `--vtt` nor `--webvtt`).

**Comment by mutschler** [comment]
you are right, currently `--skip-existing` doesn't have the intended effect when using with `-s`.
From what i see there are multiple options to change this:

1. `--skip-existing` checks for existence of a image with the same name as the source file, and skipps the whole file processing when it's found (that's how it's done when running without `-s`) 
2. Process the file and just check if the image currently written (aka current step) is already present and don't overwrite it this would result in some kind of append mode when running it multiple times with different arguments (but will result in wrong images if you increase/decrese the number of captures)

the original intention behind the `--skip-existing` flag was to skip the whole file if it already has been processed before, so you _could_ run the same command on a folder of files and only newly added files will be processed. Therefore i would go with the first approach from above: Check if ANY image exists for that file and skip the entire file.

Regarding vtt: yes, there is no vtt file created when using `-s` that's becuase when it was implemented it was based on a videojs plugin (i think) which required a sprite with image positions to display previews at the given timestamp. i honestly didn't even think about trying it with single images because while it _might_ work to use single images for each preview that would result in much more requests and loading time. 

So long story short: no vtt being generated when running in `-s` mode is indeed the intended behaviour. 

`--webvtt` is basically short for `--vtt --header=false --header-meta=false --disable-timestamps=true --padding=0` resulting in a image like [this](https://github.com/mutschler/mt/blob/master/example/mt_alloff.jpg)

**Comment by mutschler** [comment]
i've published a fix which adds support for usage of those flags together by checking for the first file and skiping the source if it is found. CI is running so the build should show up soon ;)
